build-tools/packages/build-cli/src/commands/release/report-unreleased.ts

```diff
@@ -0,0 +1,280 @@
+
+/**
+ * Fetches the dev version number released in the build feed.
+ * @param authHeader - Authorization header for Azure DevOps
+ * @param registryUrl - ADO Registry URL
+ * @param packageName - Name of the package
+ * @param buildNumber - The build number
+ * @returns The dev version number if found, otherwise undefined.
+ */
+async function fetchDevVersionNumber(
+	authHeader: string,
+	registryUrl: string,
+	packageName: string,
+	buildNumber: string,
+	log?: Logger,
+): Promise<string | undefined> {
+	try {
+		const response = await fetch(`${registryUrl}/${packageName}`, {
+			headers: { Authorization: authHeader },
+		});
+		const data = await response.json();
+		const buildVersionKey: string | undefined = Object.keys(data.time).find((key) =>
+			key.includes(buildNumber),
+		);
+
+		if (buildVersionKey !== undefined) {
+			return buildVersionKey;
+		}
+
+		log?.log(`No version with build number ${buildNumber} found.`);
+		return undefined;
+	} catch (error: unknown) {
+		log?.errorLog("Error fetching dev version number:", error);
+		return undefined;
+	}
+}
+
+/**
+ * Generates a modified manifest file with the specified version number.
+ * @param gitHubUrl - GitHub Release API URL.
+ * @param version - The version number.
+ */
+async function generateReleaseReportForUnreleasedVersions(
+	gitHubUrl: string,
+	version: string,
+	log?: Logger,
+): Promise<PackageVersionList | undefined> {
+	try {
+		const releasesResponse = await fetch(gitHubUrl);
+		const releases = await releasesResponse.json();
+
+		let manifestAsset;
+
+		for (const asset of releases[0].assets) {
+			/**
+			 * Only `caret` manifest file is required by partner repository.
+			 * dev/prerelease versions are mapped to a single version instead of ranges but other packages such as common-utils and common-definitions
+			 * required caret versions which simple manifest files do not provide.
+			 * Example of simple manifest file:
+			 * ```json
+			 * {
+			 * 	"@fluidframework/cell": "2.0.0-internal.6.1.1",
+			 *	"@fluidframework/common-definitions": "0.20.1",
+			 *	"@fluidframework/common-utils": "1.1.1",
+			 * }
+			 * ```
+			 * Example of caret manifest file:
+			 * ```json
+			 * {
+			 * 	"@fluidframework/cell": ">=2.0.0-internal.6.1.1 <2.0.0-internal.7.0.0",
+			 *	"@fluidframework/common-definitions": "^0.20.1",
+			 *	"@fluidframework/common-utils": "^1.1.1",
+			 * }
+			 * ```
+			 */
+			const includesCaretJson: boolean = asset.name.includes(".caret.json");
+			if (includesCaretJson) {
+				manifestAsset = asset;
+				break;
+			}
+		}
+
+		if (Object.keys(manifestAsset).length > 0) {
+			const manifest_url_caret = manifestAsset.browser_download_url;
+			const manifestResponse = await fetch(manifest_url_caret);
+			const manifestData = await manifestResponse.buffer();
+
+			const manifestFile: PackageVersionList = JSON.parse(manifestData.toString());
+
+			for (const key of Object.keys(manifestFile)) {
+				if (isInternalVersionRange(manifestFile[key])) {
```

Comment:
```suggestion
				if (isInternalVersionRange(manifestFile[key], true)) {
```

I think you need to pass true if you want versions like 2.0.0-dev.6.1.2.12345 to return true.